### ISSUE TITLE
Music disc fails to write URL in multiplayer servers

### ISSUE THREAD

**Issue by NatoBoram** [Plompi/VinURL#1]
The GUI fails to write music discs in multiplayer servers sometimes. There's no logs so I can't really debug it more than that.

Version used: 1.1.1

To reproduce:

1. Set an URL using the GUI
2. Play the music to completion
3. Take back the music disc and try to write it again

**Comment by Plompi** [comment]
Thanks for the bug-report, sadly your steps to reproduce the issue didnt work for me, which makes it hard to work on this issue. Because of that, this issue is on hold for now until more is known about it

**Comment by NGGiovannii** [comment]
I had the same problem but I think I found the solution, when I press esc to place the link it plays normally.

**Comment by Plompi** [comment]
I don't see how that could have anything to do with it. The logic is exactly the same for the Escape/Enter key. Could you perhaps explain in more detail what you did to solve the problem?

**Comment by NGGiovannii** [comment]
The escape key being the only way to assign the url could actually be another mod conflicting with it, but after trying to play the disc with my friend, I could only write the disc once I disconnected and connected again on the server. I have tried the mod both on lan and on Aternos, the reconnecting method worked on Aternos, but I'm not sure if that would be the same on lan.

**Comment by joacofgamer** [comment]
I'm having this issue in my minecraft server with my friends, we run on 1.21 fabric and i've tried both the 1.2.2 version and the newly released 1.2.3. The issue happens ONLY in servers, when i try it in singleplayer it always works. However in multiplayer you can only write a URL to the disc once, and after that you can't change it no matter what. The only workaround we found is that you have to relog into the server and then you can change the url 1 more time.

**Comment by Plompi** [comment]
Can only the player who set the url not change it again or every other player too?

**Comment by joacofgamer** [comment]
Yeah, i forgot to mention that. Once someone gives it a URL, no one, including the person that originally set the URL can change it. For example, i crafted a brand new disc and gave it a URL, that disc plays normally in a jukebox, but as soon as i try to change the URL again, it doesn't work. Even if i give the disc to someone else, no one can change it. We also tried /vinurl but it never worked either.

**Comment by Plompi** [comment]
I have set up a test server again and I can't reconstruct the problem. I really don't understand why. Is it possible to make a video, maybe it's something that doesn't seem worth mentioning in a textual guide

**Comment by joacofgamer** [comment]
I just got on the server by myself to record a video of the issue and magically it doesnt happen anymore?! It just works now, later maybe if my friends are online i'll test and see if it happens when other people are on the server. Really weird
